frontend/src/components/editor/file-tree/upload.tsx

```diff
@@ -40,14 +40,24 @@ export function useFileExplorerUpload(options: DropzoneOptions = {}) {
     },
     onDrop: async (acceptedFiles) => {
       for (const file of acceptedFiles) {
+        // We strip the leading slash since File.path can return
+        // `/path/to/file`.
+        const filePath = stripLeadingSlash(getPath(file));
+        let directoryPath = "" as FilePath;
+        if (filePath) {
+          directoryPath = PathBuilder.guessDeliminator(filePath).dirname(
+            filePath as FilePath,
```

Comment:
Think we can remove this cast if we update the utility function signatures below.

```suggestion
            filePath,
```